### PR TITLE
Fix openai-gpt ROCStories example's issues with AdamW optimizer

### PR DESCRIPTION
**Pull request by prrao87**
Fixes the `AdamW` optimizer instance in the `openai-gpt` ROCStories example as per the new API. The default arguments for it are now set as per the [documentation](https://huggingface.co/pytorch-transformers/model_doc/bert.html?highlight=adamw#pytorch_transformers.AdamW).

### CODE REVIEW

**Comment by thomwolf** [comment]
We should also probably add a linearly decreasing schedule (not in the optimizer anymore).
Did you try this version with flat learning rate? Does it have good performances?

**Comment by prrao87** [comment]
I didn't run the evaluation yet but I looked at it now and it seems there's an issue with the eval routine (it crashes with an error). Will look into this as soon as I have time.

**Comment by prrao87** [comment]
@thomwolf I used the code from #845 and the flat learning rate does the trick - eval accuracy of 87.2% after 3 training epochs. The defaults args in `run_openai_gpt.py` are good, just confirming some of them used for this result below. 

```
python run_openai_gpt.py  \
  --model_name openai-gpt  \
  --do_train  \
  --do_eval  \
  --train_dataset "./ROCStories/cloze_test_val__spring2016 - cloze_test_ALL_val.csv"  \
  --eval_dataset "./ROCStories/cloze_test_test__spring2016 - cloze_test_ALL_test.csv"  \
  --train_batch_size 8  \
  --eval_batch_size 16  \
  --num_train_epochs  3
```
It makes sense to go ahead and close this.  Thanks!

**Comment by thomwolf** [comment]
When I didn't have time to merge it before you closed it...

**Comment by prrao87** [comment]
Well I didn't want to take credit for the fix made in PR #845 and I didn't
really add anything to that code, so the version from #845 that you merged
to master should be the correct one. Thanks!
